python/src/uagents/context.py

```diff
@@ -329,249 +336,136 @@ def get_agents_by_protocol(
-        timeout: int = DEFAULT_ENVELOPE_TIMEOUT_SECONDS,
-    ) -> MsgStatus:
+    def _is_valid_interval_message(self, schema_digest: str) -> bool:
         """
-        Send a raw message to the specified destination.
+        Check if the message is a valid interval message.
 
         Args:
-            destination (str): The destination name or address to send the message to.
-            json_message (JsonStr): The JSON-encoded message to be sent.
-            schema_digest (str): The schema digest of the message.
-            message_type (Optional[Type[Model]]): The optional type of the message being sent.
-            sync (bool): Whether to send the message synchronously or asynchronously.
-            timeout (Optional[int]): The optional timeout for sending the message, in seconds.
+            schema_digest (str): The schema digest of the message to check.
 
         Returns:
-            MsgStatus: The delivery status of the message.
+            bool: Whether the message is a valid interval message.
         """
-        self._session = self._session or uuid.uuid4()
+        if self._interval_messages:
+            return schema_digest in self._interval_messages
+        return True
 
-        # Check if this message is a reply
-        if (
-            self._message_received is not None
-            and self._replies
-            and schema_digest != ERROR_MESSAGE_DIGEST
-        ):
-            received = self._message_received
-            # Ensure the reply is valid
-            if (received.schema_digest in self._replies) and (
-                schema_digest not in self._replies[received.schema_digest]
-            ):
-                self._logger.exception(
-                    f"Outgoing message {message_type or ''} "
-                    f"is not a valid reply to {received.message}"
-                )
-                return MsgStatus(
-                    status=DeliveryStatus.FAILED,
-                    detail="Invalid reply",
-                    destination=destination,
-                    endpoint="",
-                )
-        # Check if this message is a valid interval message
-        if (
-            self._message_received is None
-            and self._interval_messages
-            and schema_digest not in self._interval_messages
-        ):
-            self._logger.exception(
-                f"Outgoing message {message_type} is not a valid interval message"
-            )
+    async def send(
+        self,
+        destination: str,
+        message: Model,
+        sync: Optional[bool] = False,
+        timeout: Optional[int] = DEFAULT_ENVELOPE_TIMEOUT_SECONDS,
+    ) -> MsgStatus:
+        """
+        This is the pro-active send method which is used in on_event and
+        on_interval methods. In these methods, interval messages are set but
+        we don't have access properties that are only necessary in re-active
+        contexts, like 'replies', 'message_received', or 'protocol'.
```

Comment:
Mind to rephrase? I'm not really sure what this docstring is telling me and since `send` is a fairly popular method, documentation should be super clear and concise

Reply:
This is an internal comment, the developer facing comments are located at the abstract base class.